cds-snc/notification-api
# Python 3.10 v2

# Summary | Résumé

#1679 but with a working docker image! 🎉  😅 

Basically - in #1679 we were still using `eventlet` - here we switch to `gevent` as was done for admin in https://github.com/cds-snc/notification-admin/pull/1400,

# Test instructions | Instructions pour tester la modification

rebuild environment (this uses Python 3.10) and test with admin.

To test the docker image:
```
docker build -f ci/Dockerfile -t api-3.10 .
docker run -p 6013:6011 api-3.10
```
You should see normal api startup logging. Then navigate to `localhost:6013` to see the status page.
(note: bound to port 6013 here since you might have api's devcontainer running and bound to the normal 6011)


# Release Instructions | Instructions pour le déploiement

We're switching gunicorn workers from eventlet to gevent . Shouldn't make much difference, but would be good to compare perf tests of k8s pods on staging before release. (shouldn't affect lambda)


# Reviewer checklist | Liste de vérification du réviseur

This is a suggested checklist of questions reviewers might ask during their
review | Voici une suggestion de liste de vérification comprenant des questions
que les réviseurs pourraient poser pendant leur examen :


- [ ] Is the code maintainable? | Est-ce que le code peut être maintenu?
- [ ] Have you tested it? | L’avez-vous testé?
- [ ] Are there automated tests? | Y a-t-il des tests automatisés?
- [ ] Does this cause automated test coverage to drop? | Est-ce que ça entraîne
      une baisse de la quantité de code couvert par les tests automatisés?
- [ ] Does this break existing functionality? | Est-ce que ça brise une
      fonctionnalité existante?
- [ ] Does this change the privacy policy? | Est-ce que ça entraîne une
      modification de la politique de confidentialité?
- [ ] Does this introduce any security concerns? | Est-ce que ça introduit des
      préoccupations liées à la sécurité?
- [ ] Does this significantly alter performance? | Est-ce que ça modifie de
      façon importante la performance?
- [ ] What is the risk level of using added dependencies? | Quel est le degré de
      risque d’utiliser des dépendances ajoutées?
- [ ] Should any documentation be updated as a result of this? (i.e. README
      setup, etc.) | Faudra-t-il mettre à jour la documentation à la suite de ce
      changement (fichier README, etc.)?
